pytorch_lightning/profiler/pytorch.py

```diff
@@ -198,23 +272,51 @@ def __init__(
         self._register: Optional[RegisterRecordFunction] = None
         self._parent_profiler: Optional[_PROFILER] = None
         self._recording_map: Dict[str, record_function] = {}
-
-        if self._export_to_chrome and self._path_to_export_trace is None:
-            rank_zero_warn(
-                "The exported trace would be saved locally as `path_to_export_trace` is None."
-                " Note: Each functions will generate its own traced file."
-            )
+        self._start_action_name: Optional[str] = None
+        self._schedule: Optional[ScheduleWrapper] = None
+
+        if _TORCH_GREATER_EQUAL_1_8_1:
+            has_schedule = "schedule" in profiler_kwargs
+            self._has_on_trace_ready = "on_trace_ready" in profiler_kwargs
+            schedule = profiler_kwargs.get("schedule", None)
+            activities = profiler_kwargs.get("activities", None)
+            if schedule is not None:
+                if not isinstance(schedule, Callable):
+                    raise MisconfigurationException(f"Found schedule: {schedule}. Schedule should be a callable.")
+                action = schedule(0)
+                if not isinstance(action, ProfilerAction):
+                    raise MisconfigurationException(
+                        f"Found schedule: {schedule}. "
+                        "Schedule should be a callable returning `torch.profiler.ProfilerAction`. "
+                    )
+
+            schedule = schedule if has_schedule else self._default_schedule()
+            self._schedule = ScheduleWrapper(schedule) if schedule is not None else schedule
+            self._profiler_kwargs["schedule"] = self._schedule
+            self._profiler_kwargs["activities"] = activities or self._default_activities()
+            self._export_to_flame_graph = profiler_kwargs.get("export_to_flame_graph", True)
+            self._metric = profiler_kwargs.get("metric", "self_cpu_time_total")
+            self._profiler_kwargs["with_stack"] = profiler_kwargs.get("with_stack", True) or self._export_to_flame_graph
```

Comment:
with_stack has performance overhead, it's not recommended this is on by default. It's there to be able to get attribution back to Python source when needed, but when enabled we won't get a representative profile.

Reply:
Thanks for the information

Reply:
Default to False